src/Enzo/enzo_EnzoMethodGrackle.cpp

```diff
@@ -412,6 +444,10 @@ void EnzoMethodGrackle::update_grackle_density_fields(
           grackle_fields->HDI_density[i]   = grackle_fields->density[i] * tiny_number;
         }
 
+       if (grackle_chemistry->metal_cooling == 1){
+          grackle_fields->metal_density[i] = grackle_fields->density[i] * enzo_config->method_grackle_metallicity_floor*cello::metallicity_solar;
```

Comment:
Is there any reason why we aren't storing `metallicity_floor` as an attribute of the `EnzoMethodGrackle` class (instead of accessing it through `enzo_config->method_grackle_metallicity_floor`? 

If there isn't one, I'd request that you make that change (since this is the convention adopted throughout the rest of the codebase).

Reply:
Finally coming back to this -- sorry for the delay. I'm actually having trouble storing `metallicity_floor` as an attribute of `EnzoMethodGrackle` and accessing it here because this particular function is defined as a `static` method (so there's no `this` pointer). Do you know if there's any particular reason why or if it's necessary for these functions to be static?

Reply:
I believe that this method is static to make the method easier to call from problem initializers. But, I think it would be fine to make this function non-static. You would just need to replace to `EnzoMethodGrackle::update_grackle_density_fields` with `enzo::grackle_method()->update_grackle_density_fields` in `enzo_EnzoInitialCosmology.cpp` and `enzo_EnzoInitialIsolatedGalaxy.cpp`

Alternatively, it's probably fine to leave to leave this as written.

Reply:
I went ahead and made this method, along with `setup_grackle_fields` and `delete_grackle_fields`, nonstatic, and called them using `enzo::grackle_method()`, since this is consistent with how other Grackle methods are being called externally elsewhere in the code (e.g. `calculate_temperature` in `EnzoComputeTemperature`)